src/cli/proxy_mode.rs

```diff
@@ -12,7 +12,7 @@ use crate::{
 };
 
 #[cfg_attr(feature = "otel", tracing::instrument)]
-pub fn main(arg0: &str) -> Result<ExitCode> {
+pub async fn main(arg0: &str) -> Result<ExitCode> {
```

Comment:
Have you checked the performance impact of these changes on proxy invocations?

Reply:
Not rigorously, but I know that most of our proxy overhead is in manifest parsing today, so I doubt that we'd get a measurable change even if the runtime initialisation were a concern.